### ISSUE TITLE
Software item type

### ISSUE THREAD

**Issue by rmzelle** [avram/zotero-bits#69]
Per https://github.com/citation-style-language/styles/pull/1704/files#r41761578

(@adam3smith)

**Comment by adam3smith** [comment]
Software is increasingly being cited
http://blog.datacite.org/software-citation-workflows/
And both Mendeley and Zotero have computer program item types. This seems like an obvious addition.

**Comment by rmzelle** [comment]
Preferred CSL item type name? "program", "computer_program", "software"? I think I prefer "program".

**Comment by bdarcus** [comment]
Program could be confusing.

http://dictionary.reference.com/browse/program

**Comment by rmzelle** [comment]
Is "software" too general? Otherwise I guess we'll go with "computer_program", then.

**Comment by adam3smith** [comment]
I actually like software, don't see how that'd be too general (if anything that's a plus), and it's nice and pithy.

**Comment by rmzelle** [comment]
Fine with me!

**Comment by slint** [comment]
@rmzelle Is there any status update (or PR in https://github.com/citation-style-language/schema) for this "software" type addition?

**Comment by adam3smith** [comment]
no -- item type additions happen in new CSL releases and those take time to prepare (no need for a PR -- the work here is not the actual change in the schema, which is trivial, but reaching agreement on what should and shouldn't be in a next release).

**Comment by rmzelle** [comment]
While I've been procrastinating for over 4 years now when it comes to working on a new CSL release, I have some good hopes we can rekindle the fire and push out a (small) update in 2017.

One reason I didn't feel much pressure is that adding new item types wasn't really feasible for Zotero 4.0.x, but this situation is about to change now that Zotero 5.0 has nearly arrived (while there are many more programs that use CSL, I didn't want to leave Zotero out in the cold, since they've been one of CSL's biggest institutional supporters). That's what I tell myself, anyways.

**Comment by njsmith** [comment]
I'm lurking on this issue because I'm hoping to see a solution for projects like [duecredit](https://github.com/duecredit/duecredit) that need a standard interchange format for citations, and in particular software citations -- right now AFAICT there really isn't any reasonable format for them to use.

**Comment by rmzelle** [comment]
@njsmith, yes, we're aware of the high interest in the community to improve software citation, so a new item type for software will almost certainly be added to the next CSL release. Can't provide a more precise ETA right now, though.

**Comment by tjhei** [comment]
I am curious if there are any updates regarding a ``software`` citation type. I found my way here from zenodo.org, which can not display the correct type and chooses to show software DOIs as "[data set]" because of this. Thanks!

**Comment by rmzelle** [comment]
Status is still the same as above.

**Comment by timjim333** [comment]
Hi, is this still a work in progress? =) Thanks! As a workaround, is this something we can pull and modify?

**Comment by denismaier** [comment]
My impression is there's been a consensus that this should be implemented, right? If yes, can we move this to accepted?

**Comment by dstillman** [comment]
We renamed Computer Program to Software a while ago in Zotero. Is there anything else we need to do here?

**Comment by bwiernik** [comment]
We need to map it to CSL 1.0.2 `software`. 

Also, I don't know that we really need both System (`medium`) and Programming Language (`genre`) fields. This sort of information is generally just cited with a single label like "Computer program", "Software", "R package", "Python package", etc. 

I don't ever see references to System (Windows, Linux, MacOS) in citations, and the Programming Language label is confusing. Should it contain, e.g., "R" or "R package" (the latter being what would be cited). What about something like an R Package that has both R and C++ code in it?

I think consolidating these fields to just "Type" or "Software Type" (mapped to `genre`) would make sense. 

Or perhaps for backwards compatibility, add "Software Type" (`genre`) and remove the CSL mappings for System and Programming Language. Perhaps migrate Programming Language to that field (and maybe System instead if Programming Language is empty).

**Comment by dstillman** [comment]
We definitely want to keep Programming Language, since it's fairly important for non-citation usage, and we save to that from GitHub (e.g., "Python"). We could consider duplicating "Programming Language" values into "Type", but I wouldn't want to just move them.

"System" should probably be "Platform" for clarity.

**Comment by dstillman** [comment]
Should we try to populate "Type" from the GitHub translator? If all we have is, say, "Python"?

The Type fields in general are kind of unfortunate — it seems like they're often tied to style requirements, which sort of breaks the fundamental model of data vs. presentation and means we can't generate correct citations out of the box…

**Comment by bwiernik** [comment]
I would suggest the following:
- Remove mapping of System and Programming Language
- Duplicate Programming Language in Type
- Rename System to Platform

I don't think that the Type fields are _that_ much of a nuisance. The only style that is quite prescriptive about the exact wording of types is APA, and even there it got a lot more flexible in terms of what is allowable in 7th edition—phrases like "computer software" are sort of like fallbacks if there isn't a more specific term (e.g., "Python package"). Other styles are really flexible about exact phrasing. 

The biggest limitation about Type/Format fields (`genre`/`medium`) at the moment I think is that they aren't localizable. That could be fixed with field-level language tags in a future version.

**Comment by bwiernik** [comment]
> Should we try to populate "Type" from the GitHub translator? If all we have is, say, "Python"?

I think populating "Type" from the GitHub and CRAN translators would be good. "Programming language" + " package" would work a huge amount of the time, as least for the commonly most cited languages (R, Python, Julia, Matlab). Do you have a sample of GitHub programming lanugage field data?
